pkg/webhook/preflight/nutanix/failuredomain.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nutanix
+
+import (
+	"context"
+	"fmt"
+
+	netv4 "github.com/nutanix/ntnx-api-golang-clients/networking-go-client/v4/models/networking/v4/config"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	capxv1 "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/external/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/api/v1beta1"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/webhook/preflight"
+)
+
+type failureDomainCheck struct {
+	failureDomainName string
+	namespace         string
+	field             string
+	kclient           ctrlclient.Client
+	nclient           client
+}
+
+func (fdc *failureDomainCheck) Name() string {
+	return "NutanixFailureDomains"
+}
+
+func newFailureDomainChecks(cd *checkDependencies) []preflight.Check {
+	checks := []preflight.Check{}
+
+	if cd.nclient == nil || cd.kclient == nil {
+		return checks
+	}
+
+	// For the failure domains configured for control-plane nodes
+	if cd.nutanixClusterConfigSpec != nil &&
+		cd.nutanixClusterConfigSpec.ControlPlane != nil &&
+		cd.nutanixClusterConfigSpec.ControlPlane.Nutanix != nil {
+		for _, fdName := range cd.nutanixClusterConfigSpec.ControlPlane.Nutanix.FailureDomains {
+			if fdName != "" {
+				checks = append(checks, &failureDomainCheck{
+					failureDomainName: fdName,
+					namespace:         cd.cluster.Namespace,
+					field: fmt.Sprintf(
+						"cluster.spec.topology.variables[.name=clusterConfig].value.controlPlane.nutanix.failureDomains[%s]",
```

Comment:
Use JSONPath syntax.
```suggestion
						"$.spec.topology.variables[?@.name==\"clusterConfig\"].value.controlPlane.nutanix.failureDomains[?@.name==%q]",
```

Reply:
changed the field as suggested